### PR TITLE
Freeplane: init at 1.6.13

### PR DESCRIPTION
**Pull request by freeman42x**
Packaged Freeplane based on the expression from Freemind. Freeplane is a project based on Freemind but made to be better in some ways.

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/freeplane/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    export GRADLE_USER_HOME="/tmp"
+    gradle dist
+  '';
+
+  installPhase = ''
+    mkdir -p $out/{bin,nix-support}
+    cp -r BIN $out/nix-support
+    sed -i 's/which/type -p/' $out/nix-support/BIN/freeplane.sh
+
+    cat >$out/bin/freeplane <<EOF
```
Comment (Mic92) [inline pkgs/applications/misc/freeplane/default.nix]: consider using `makeWrapper` here.

Comment (razvan-panda) [reply pkgs/applications/misc/freeplane/default.nix]: @Mic92 I found https://nixos.org/nixpkgs/manual/#ssec-stdenv-functions but I do not understand how to use it in this case.

Comment (razvan-panda) [reply pkgs/applications/misc/freeplane/default.nix]: @puffnfresh explained to me how it works. I'll update as soon as Freeplane .desktop PR is done.

Comment (razvan-panda) [reply pkgs/applications/misc/freeplane/default.nix]: @Mic92 Updated to use `makeWrapper` and also to use .desktop file

**Comment by Mic92** [comment]
@GrahamcOfBorg build freeplane

**Comment by Mic92** [comment]
The error you see here, is related to the fact, that gradle tries to download something.
Network access is not allowed if nix sandboxes are enabled.

**Comment by Mic92** [comment]
I think the common solution for most java-based projects is to use `jar` files provided by upstream.
We do not have good nix tooling for java projects.

**Comment by razvan-panda** [comment]
@Mic92 Can this be merged or what else is required?

**Comment by joachifm** [comment]
@razvan-panda I expect `gradle dist` still requires network access during the build? If so, you must find a way to make the dependencies `dist` acquires explicit. Usually there's some form of tooling that parses dependency information and dumps corresponding Nix expressions; per @Mic92's comment I suppose you'll need to do this by hand somehow.

I seem to recall that @NeQuissimus has done some work with gradle in the past, they might know more.

**Comment by razvan-panda** [comment]
@joachifm Thank you for feedback. I'll try to use https://github.com/mcpkg/gradle2nix and see if it still works.

**Comment by razvan-panda** [comment]
gradle2nix does not currently build and it is abandoned: https://github.com/mcpkg/gradle2nix/issues/1

**Comment by razvan-panda** [comment]
I got it working but is seem so be more like pom2nix instead of gradle2nix: https://github.com/razvan-panda/gradle2nix

**Comment by puffnfresh** [comment]
Still working on this @razvan-panda ?

**Comment by razvan-panda** [comment]
@puffnfresh Not working on it anymore since I could not find any way to solve it. I asked around quite a bit but no one knew how to do this either.

**Comment by aanderse** [comment]
@razvan-panda the state of java in nixpkgs is rather unfortunate but if you intend to continue with this PR your best bet is to use the `-bin` release that upstream provides so you don't need to compile any jar files.

That being said... any motivation to continue with this PR?

**Comment by razvan-panda** [comment]
@aanderse I've abandoned this PR since it is low on my priority list.

**Comment by aanderse** [comment]
@razvan-panda thanks for the update. Maybe someone will pick this up at some point in the future... :man_shrugging:

**Pull request closed without merging**